lib/dataset/dataset.cpp

```diff
@@ -163,14 +178,13 @@ void Dataset::setData(const std::string &name, DataArray data) {
         it->attrs() == data.attrs() && it->coords() == data.coords())
       return;
   }
-  for (auto &&[dim, coord] : data.coords())
-    if (const auto it = m_coords.find(dim); it != m_coords.end())
-      expect::matching_coord(dim, coord, it->second, "set coord");
+  const auto to_skip = coords_to_skip(*this, data);
 
   setData(name, data.data());
   auto &item = m_data[name];
   for (auto &&[dim, coord] : data.coords())
-    if (const auto it = m_coords.find(dim); it == m_coords.end())
+    if (m_coords.find(dim) == m_coords.end() &&
+        std::find(to_skip.begin(), to_skip.end(), dim) == to_skip.end())
```

Comment:
Instead of setting up`to_skip` and handling it here, can't you just drop coords from `data`? It is taken by value anyway.

Reply:
I split it up like this because we can't erase from the coords while iterating over them. I could change it to erase the coords *after* setting up `to_skip` instead of the current mechanism using `find`. But is this actually better?